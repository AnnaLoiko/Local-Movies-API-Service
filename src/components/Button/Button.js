import React, { useEffect } from "react";
import style from "./style.module";

import PropTypes from 'prop-types';

const Button = React.memo((props) => {
  const { type = "text", text = "Button", className = "btnSecondary", addClassName, disabled = false, handleClick, autoFocus } = props;


  return (
    <button
      type={type}
      className={`${style[className]} ${addClassName && style[addClassName]} `}
      disabled={disabled}
      onClick={handleClick}
      autoFocus={autoFocus}
    >
      <span>{text}</span>
    </button>
  )
});


Button.propTypes = {
  type: PropTypes.string,
  text: PropTypes.string,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  handleClick: PropTypes.func,
};

export default Button;
