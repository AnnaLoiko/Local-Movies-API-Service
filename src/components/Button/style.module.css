.btnPrimary {
    display: inline-block;
    color: #ffffff;
    background-color: #F65261;
    border: 2px solid #F65261;
    border-radius: 5px;
    text-align: center;
    text-transform: uppercase;
    padding: 0 30px;
    font-family: 'Roboto', Arial, Helvetica, sans-serif;
    font-size: 20px;
    font-weight: 400;
    letter-spacing: .5px;
    line-height: 54px;
    vertical-align: top;
    min-width: 180px;
}

.btnPrimaryInvert {
    display: inline-block;
    background-color: #232323;
    color: #F65261;
    border: 2px solid #F65261;
    border-radius: 5px;
    text-align: center;
    text-transform: uppercase;
    padding: 0 30px;
    font-family: 'Roboto', Arial, Helvetica, sans-serif;
    font-size: 20px;
    font-weight: 400;
    letter-spacing: .5px;
    line-height: 54px;
    vertical-align: top;
    min-width: 180px;
}

.btnSecondary {
    display: inline-block;
    color: #F65261;
    background: rgba(66,66,66,0.8);
    border: 2px solid rgba(66,66,66,0.8);
    border-radius: 5px;
    text-align: center;
    font-size: 22px;
    line-height: 50px;
    text-transform: uppercase;
    padding: 0 20px;
    font-weight: 500;
    font-family: 'Roboto', Arial, Helvetica, sans-serif;
}

.btnLarge {
    min-width: 240px;
}

.btnZoom {
    width: 60px;
    height: 40px;
    background-image: url("../../assets/images/zoom.png");
    background-repeat: no-repeat;
    background-size: 30px;
    background-position: center;
}

.btnPrimary:hover,
.btnPrimary:active,
.btnPrimaryInvert:hover,
.btnPrimaryInvert:active,
.btnSecondary:hover,
.btnSecondary:active {
    border: 2px solid #efefef;
    color: #efefef;
}

.btnPrimary:disabled,
.btnSecondary:disabled {
    border: 2px solid #efefef;
    color: #efefef;
}


