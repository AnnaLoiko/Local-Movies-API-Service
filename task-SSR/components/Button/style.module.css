.btnPrimary {
    display: inline-block;
    color: #ffffff;
    background-color: #F65261;
    border: 2px solid #F65261;
    border-radius: 5px;
    text-align: center;
    text-transform: uppercase;
    padding: 0 30px;
    font-family: 'Roboto', Arial, Helvetica, sans-serif;
    font-size: 20px;
    font-weight: 400;
    letter-spacing: .5px;
    line-height: 54px;
    vertical-align: top;
    min-width: 180px;
}

.btnPrimaryInvert {
    display: inline-block;
    background-color: #232323;
    color: #F65261;
    border: 2px solid #F65261;
    border-radius: 5px;
    text-align: center;
    text-transform: uppercase;
    padding: 0 30px;
    font-family: 'Roboto', Arial, Helvetica, sans-serif;
    font-size: 20px;
    font-weight: 400;
    letter-spacing: .5px;
    line-height: 54px;
    vertical-align: top;
    min-width: 180px;
}

.btnSecondary {
    display: inline-block;
    color: #F65261;
    background: rgba(66,66,66,0.8);
    border: 2px solid rgba(66,66,66,0.8);
    border-radius: 5px;
    text-align: center;
    font-size: 22px;
    line-height: 50px;
    text-transform: uppercase;
    padding: 0 20px;
    font-weight: 500;
    font-family: 'Roboto', Arial, Helvetica, sans-serif;
}

.btnLarge {
    min-width: 240px;
}

.btnPrimary:hover,
.btnPrimary:active,
.btnSecondary:hover,
.btnSecondary:active {
    border: 2px solid #fc626f;
    background-color: #fc626f;
    color: #efefef;
}

.btnPrimaryInvert:hover,
.btnPrimaryInvert:active{
    border: 2px solid #ac3a44;
    background-color: #000000;
    color: #ac3a44;
}

.btnPrimary:disabled,
.btnSecondary:disabled {
    border: 2px solid #353434;
    background-color: #353434;
    color: #efefef;
}