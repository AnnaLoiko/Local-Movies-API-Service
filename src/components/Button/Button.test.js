import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from "enzyme-adapter-react-16";
import Button from './Button';

Enzyme.configure({ adapter: new Adapter() });


describe('Testing Button component:', () => {
  it('Expects to find button HTML element in the DOM', () => {
    const elem = renderer.create(<Button />).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('Test click button event', () => {
    const mockCallBack = jest.fn();

    const button = shallow((<Button handleClick={mockCallBack}>Ok!</Button>));
    button.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});