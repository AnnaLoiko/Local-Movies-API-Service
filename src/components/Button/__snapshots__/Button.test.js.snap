// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component: Expects to find button HTML element in the DOM 1`] = `
<button
  autoFocus={false}
  className="btnSecondary  "
  disabled={false}
  type="text"
>
  <span>
    Button
  </span>
</button>
`;
